js/src/data/actions.js

```diff
@@ -765,14 +765,85 @@ export function receiveAdsCampaigns( adsCampaigns ) {
 	};
 }
 
-export function* updateAdsCampaign( id, data ) {
+/**
+ * Create a new ads campaign.
+ *
+ * @param {number} amount Daily average cost of the paid ads campaign.
+ * @param {string} country Country code of the paid ads campaign audience country. Example: 'US'.
+ *
+ * @throws { { message: string } } Will throw an error if the campaign creation fails.
+ */
+export function* createAdsCampaign( amount, country ) {
 	try {
-		yield apiFetch( {
-			path: `/wc/gla/ads/campaigns/${ id }`,
+		const createdCampaign = yield apiFetch( {
+			path: `${ API_NAMESPACE }/ads/campaigns`,
 			method: 'POST',
+			data: {
+				amount,
+				country,
+			},
+		} );
+
+		{
+			/**
+			 * If the campaign creation fails, the API still responds a HTTP status code 200 response
+			 * with `message` property only. So here check the fails by data structure.
+			 *
+			 * TODO: Remove this code block after the API responds the fails in a more specific way.
+			 */
+			const keys = Object.keys( createdCampaign );
+			if ( keys.length === 1 && keys.pop() === 'message' ) {
```

Comment:
I was leaning toward adjusting them after the #1210 is fixed and not to refactor workarounds based on uncertain future changes for two reasons:
1. They are using different APIs and the APIs are also independent implementations.
2. I assumed the fixing direction of APIs would be more likely to respond with HTTP status code 400 or something like that. And then we just need to delete the whole block without having to check other function references and whether that functions are shared elsewhere.

Reply:
> And then we just need to delete the whole block without having to check other function references and whether that functions are shared elsewhere.

If we're going to use the function elsewhere, we shouldn't delete it. We just need to delete/update the line calling the function once the API response is fixed. Besides that, I think is harder to locate the block duplicated over the code base than a named function. 

 > They are using different APIs and the APIs are also independent implementations.

In that case, is more likely to happen that one of them is fixed and then the other implementation is fixed after. Hence makes more sense for me to have a temporary function.  

> I was leaning toward adjusting them after the #1210 is fixed

I'm fine having this refactor in the future since this is not a blocker. But I think would be more correct to have it in a function :)

Reply:
Sorry, I don't really understand your thoughts.

> If we're going to use the function elsewhere, we shouldn't delete it. We just need to delete/update the line calling the function once the API response is fixed.

This temporary workaround is not intended to reuse elsewhere. And I don't see why the code needs to be deleted is not clear enough. There are code comments and **TODO** flag, and the codes are wrapped by a block.

https://github.com/woocommerce/google-listings-and-ads/blob/92b041a406d016c539c86bb6562e3f04f83dfdd4/js/src/data/actions.js#L787-L798

Would it more clearer for you if I add those code blocks to #1210 once this PR gets merged?

Reply:
At the beginning, it was suggested to refactor these blocks in a function i.e `temporaryCheckFunctionResponse(createdCampaign)`

One of the reasons mentioned to don't do so was this:

> 2. I assumed the fixing direction of APIs would be more likely to respond with HTTP status code 400 or something like that. And then we just need to delete the whole block without having to check other function references and whether that functions are shared elsewhere.

Maybe I understood it wrong. But this is what I get for that sentence:

- The code is going to be deleted or modified 
- The code is going to be harder to delete or modify if we refactor it as a function. Because we would need to check function references and whether those functions are shared elsewhere -> This part I disagree... since it would be easier the modification in the form of a function than a duplicated block as is now.

In brief, my thoughts are that right now, the code is duplicated and could be refactored.  I understand the intent of the code and its temporary nature. But I disagree about that temporary nature is a justification to have it duplicated.

Reply:
Thanks for the explanation.

> * The code is going to be deleted or modified
> * [...] This part I disagree... since it would be easier the modification in the form of a function than a duplicated block as is now.

Based on most of current APIs will respond with status code 400 when parameters are invalid. I assumed most likely the code block would be deleted, not modified. Even if the error is later specified in a different way, how it should be refactored would be clearer at that point. Considering whether to refactor it now seems to be a bit over-engineering to me.

In addition, a small amount of a few lines of duplicated code does not always have a bad smell, sometimes it brings nicer readability to code fragments, here I think the main consideration is how much it burdens the reader's mind, more than whether it is very much aligned with the clean code styles, DRY concepts, or whether it is easy to maintain, since it would most probably be deleted in its entirety.

> In brief, my thoughts are that right now, the code is duplicated and could be refactored. I understand the intent of the code and its temporary nature. But I disagree about that temporary nature is a justification to have it duplicated.

I would like to wrap up this thread with the current implementation and proceed with PR merging. The above are mainly my subjective thoughts, but not a strong opinion. And I know the code is not ideal in some perspectives. I would be happy to adopt your proposed implementation, but I have no idea what's the ideal implementation you thought to refactor it. If you think it's necessary, I would appreciate it if you could open a PR for it. 🙏